manifests/stage-ib-kubernetes/0040-deployment.yaml

```diff
@@ -0,0 +1,124 @@
+# Copyright 2022 NVIDIA CORPORATION & AFFILIATES
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: ib-kubernetes
+  namespace: {{ .RuntimeSpec.Namespace }}
+  annotations:
+    kubernetes.io/description: |
+      This deployment launches the ib-kubernetes daemon for Infiniband CNI.
+spec:
+  progressDeadlineSeconds: 600
+  strategy:
+    type: RollingUpdate
+  replicas: 1
+  selector:
+    matchLabels:
+      name: ib-kubernetes
+  template:
+    metadata:
+      labels:
+        name: ib-kubernetes
+        component: network
+        type: infra
+        kubernetes.io/os: "linux"
+    spec:
+      priorityClassName: system-cluster-critical
+      serviceAccountName: ib-kubernetes
+      hostNetwork: true
+      # required to be scheduled on a linux node with node-role.kubernetes.io/master label and
+      # only one instance of ib-kubernetes pod per node
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+              - matchExpressions:
+                  - key: node-role.kubernetes.io/master
+                    operator: In
+                    values:
+                      - ""
+                  - key: kubernetes.io/os
+                    operator: In
+                    values:
+                      - "linux"
+              - matchExpressions:
+                  - key: node-role.kubernetes.io/control-plane
+                    operator: In
+                    values:
+                      - ""
+                  - key: kubernetes.io/os
+                    operator: In
+                    values:
+                      - "linux"
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            - labelSelector:
+                matchExpressions:
+                  - key: name
+                    operator: In
+                    values:
+                      - ib-kubernetes
+              topologyKey: kubernetes.io/hostname
+      containers:
+        - name: ib-kubernetes
+          image: {{ .CrSpec.Repository }}/{{ .CrSpec.Image }}:{{ .CrSpec.Version }}
+          imagePullPolicy: IfNotPresent
+          command: ["/usr/bin/ib-kubernetes"]
+          resources:
+            requests:
+              cpu: 100m
+              memory: 300Mi
+          env:
+            - name: DAEMON_SM_PLUGIN
+              value: "ufm"
```

Comment:
Should it be configurable via CRD?

Reply:
There's only one plugin implemented at the moment, I don't think it needs to be configured at the moment.

Reply:
ok, sounds reasonable